lib/AST/Decl.cpp

```diff
@@ -3750,7 +3750,7 @@ void VarDecl::emitLetToVarNoteIfSimple(DeclContext *UseDC) const {
   // since the user has to choose to apply this anyway.
   if (auto *PBD = getParentPatternBinding()) {
     // Don't touch generated or invalid code.
-    if (PBD->getLoc().isInvalid() || PBD->isImplicit())
+    if (PBD->getLoc().isInvalid() || PBD->isImplicit() || isCaptureList())
```

Comment:
You don't need `PBD` here, you can hoist this out.
Maybe, right after [`if (!isLet()) return;` line](https://github.com/apple/swift/pull/6490/files#diff-18d4a886db0259206b4f3bd502d67c75R3726):
```cpp
// Don't touch capture list
if (isCaptureList()) return;
```